app/components/ui/select-data-modal/component.js

```diff
@@ -0,0 +1,205 @@
+
+export default Ember.Component.extend({
+  userAuth: service(),
+  folderNavs: service(),
+  store: service(),
+
+  selectedMenuIndex: 0,
+  dataSources: A([
+    O({name: 'WholeTale Catalog'}),
+    O({name: 'Tale Workspaces'})
+  ]),
+  selectedDataSource: O({}),
+
+  allSelectedItems: A(),
+
+  folders: A(),
+  files: A(),
+
+  loading: true,
+  currentFolder: null, 
+  rootFolderId: null,
+
+  model: O({}),
+  classNameBindings: ['injectedClassName'],
+
+  injectedClassName: Ember.computed('modelType', 'model._modelType', 'model._id', function () {
+    if(this.get('modelType')) {
+        let newClass = `select-data-modal-${this.get('modelType')}`;
+        return newClass;
+    } else return '';
+  }),
+
+  actions: {
+    selectDatasource(datasource, index) {
+      this.set('selectedDataSource', datasource);
+      this.set('selectedMenuIndex', index);
+    },
+
+    initData() {
+      this.selectedDataSource = this.get('dataSources')[this.get('selectedMenuIndex')];
+      this.initData.call(this);
+    },
+
+    updateSessionData() {
+      this.updateSessionData.call(this, this.get('allSelectedItems'));
+    },
+
+    cancel() {
+      this.cancel.call(this);
+    },
+
+    dblClick(target) {
+      this.dblClick.call(this, target);
+    },
+
+    onClick(target) {
+      this.onClick.call(this, target);
+    },
+
+    goBack() {
+      this.goBack.call(this, this.get('currentFolder'));
+    },
+
+    addSelectedData() {
+      this.addSelectedData.call(this, this.get('files'), this.get('folders'));
+    },
+
+    removeSelectedData() {
+      this.removeSelectedData.call(this);
+    }
+  },
+
+  // -----------------------------------------------------------------------
+  // BELOW ARE FUNCTIONS FOR DEFAULT BEHAVIOR: 
+  //    You can set custom behavior by overriding these functions. 
+  // -----------------------------------------------------------------------
+
+  initData() {
+    this.set('loading', true);
+
+    this.set('allSelectedItems', A());
+    this.set('folders', A());
+    this.set('files', A());
+    this.set('currentFolder', null);
+    this.set('rootFolderId', null);
+
+    const store = this.get('store');
+    let parentId = this.get('userAuth').getCurrentUserID();
+    let dataNavInfo = this.get('folderNavs').getFolderNavFor('user_data');
+    let parentType = dataNavInfo.parentType;
+    let name = dataNavInfo.name;
+    let adapterOptions = {queryParams: {limit: "0"}};
+    
+    const self = this;
+    return store.query('folder', {parentId, parentType, name, adapterOptions}).then(dataFolder => {
+      let dataFolderId = dataFolder.content[0].id;
+      let parentCollection = parentType;
+      self.set('rootFolderId', dataFolderId);
+      self.set('currentFolder', O({id: dataFolderId, _modelType: 'folder', parentCollection, parentId}));
+
+      return self.loadFolder.call(self, dataFolderId, 'folder');
+    }).catch(e => {
+      self.set('loading', false);
+      console.error(e);
```

Comment:
If we reach this state, will any files/folders be displayed? If not we might want to show an error to the user.
Something along the lines of `Failed to load registered data. Please try again`.

Reply:
I agree